tools/hermetic_grunt/Dockerfile

```diff
@@ -0,0 +1,54 @@
+# To build and run the hermetic grunt:
+# docker build -f tools/hermetic_grunt/Dockerfile -t uproxy-grunt . && \
+# docker run --rm -v $(pwd):/root/repository_root -it uproxy-grunt
+#
+# You have to run from the repository root. The output is written to the out/
+# directory.
+
+FROM ubuntu
+
+LABEL description="Docker container for hermetic grunt runs"
+
+###############################################################################
+# Install prerequisite global tools
+###############################################################################
+
+RUN apt-get update
+
+RUN apt-get install -qqy git python build-essential
+
+# Add a fresher repository for nodejs
+RUN apt-get install curl && \
+  curl -sL https://deb.nodesource.com/setup_4.x | bash - && \
+  apt-get install -qqy nodejs && \
+  apt-get purge -y --auto-remove curl
```

Comment:
Any reason to uninstall `curl`? Since you have to download it in the first place I think this only increases image build time. Also, `curl` is a key tool for interacting with our sample apps such as Simple SOCKS.